packages/typescript-estree/src/parseSettings/ExpiringCache.ts

```diff
@@ -0,0 +1,111 @@
+export const DEFAULT_TSCONFIG_CACHE_DURATION_SECONDS = 30;
+
+/**
+ * A map with key-level expiration.
+ */
+export class ExpiringCache<TKey, TValue> {
+  readonly #cacheDurationSeconds: number | 'Infinity';
```

Comment:
This map has to check `this.#cacheDurationSeconds` every time there's a `.get` or `.set`. How about switching to two different maps, with a factory-ish function between them?

```ts
interface CacheMap { ... }

// number case
class ExpiringCacheMap implements CacheMap { ... }

function getCacheMap(duration: number | 'Infinity') {
  return duration === 'Infinity' ? new Map() : new ExpiringCacheMap(duration);
}
```

Reply:
![](https://media.tenor.com/n6giORGdzL0AAAAC/larry-david-well.gif)

That's a nice idea. But I think i'll leave it as is - just saves the layer of indirection at the top-level and keeps the store a consistent type.
If we allow a dynamic duration in future it saves us undoing this change.